dash/include/dash/allocator/CollectiveAllocator.h

```diff
@@ -123,12 +126,15 @@ class CollectiveAllocator
   /**
    * Move-assignment operator.
    */
-  self_t & operator=(const self_t && other) noexcept
+  self_t & operator=(self_t && other) noexcept
   {
     // Take ownership of other instance's allocation vector:
     if (this != &other) {
-      std::swap(_allocated, other._allocated);
```

Comment:
Are you really sure about that: If `_allocated` is not empty, the elements are moved to `other._allocated`. This is especially problematic, as the deallocator of `other` will free them later. The call to `other._allocated.clear()` is necessary, as the move assignment operator of `std::vector` does not ensure that the elements are freed. We have to ensure that however to avoid deallocation in the destructor. My statement is based on the documentation on [cppreference / `std::vector`](http://en.cppreference.com/w/cpp/container/vector/operator%3D). Maybe I missed something.